### DISCUSSION TITLE
Regarding custom vc template addition

### DISCUSSION THREAD

**Discussion by SethiShivam** [walt-id/.github#19] [Help] [answered]
Hi Team ,
I have a question , 

Do we need to follow this procedure everytime  , if we want to create a new Vc template

**Adding VC Templates
Visit https://vc-creator.walt.id/ to generate a VC template.
Add it to package: id.walt.vclib.credentials.*, compare it to the other templates and complete the implementation.
Add the new class to id.walt.vclib.registry.VcTypeRegistry.
Run all test-cases, which will generate the JsonSchema and the serialized version of the credential template.
Open a Pull-Request on GitHub**

or is there any way that we can simply create a custom template while doing testing ?

Best Regards
Sethi Shivam

**Comment by SethiShivam** [comment]
@philpotisk  @taminobaumann

**Comment by philpotisk** [comment, marked as answer]
Well, currently you would need to add the custom credential template in code as shown here https://github.com/walt-id/waltid-ssikit-examples/blob/master/src/main/kotlin/id/walt/ssikitexamples/CustomCredential.kt

In case you want to make your credential publicly available please feel free to add this to the VcLib https://github.com/walt-id/waltid-ssikit-vclib via a PR.

Please also be aware that we will restructure the lib in future so you will have the option to don't use typed credentials if you don't want to.
